packaging/installer/functions.sh

```diff
@@ -191,6 +191,112 @@ netdata_banner() {
   echo >&2
 }
 
+# -----------------------------------------------------------------------------
+# Feature management and configuration commands
+
+enable_feature() {
+  NETDATA_CMAKE_OPTIONS="$(echo "${NETDATA_CMAKE_OPTIONS}" | sed -e "s/-DENABLE_${1}=Off[[:space:]]*//g" -e "s/-DENABLE_${1}=On[[:space:]]*//g")"
+  if [ "${2}" -eq 1 ]; then
+    NETDATA_CMAKE_OPTIONS="$(echo "${NETDATA_CMAKE_OPTIONS}" | sed "s/$/ -DENABLE_${1}=On/")"
+  else
+    NETDATA_CMAKE_OPTIONS="$(echo "${NETDATA_CMAKE_OPTIONS}" | sed "s/$/ -DENABLE_${1}=Off/")"
+  fi
+}
+
+check_for_module() {
+  if [ -z "${pkgconf}" ]; then
+    pkgconf="$(command -v pkgconf 2>/dev/null)"
+    [ -z "${pkgconf}" ] && pkgconf="$(command -v pkg-config 2>/dev/null)"
+    [ -z "${pkgconf}" ] && fatal "Unable to find a usable pkgconf/pkg-config command, cannot build Netdata." I0013
+  fi
+
+  "${pkgconf}" "${1}"
+  return "${?}"
+}
+
+check_for_feature() {
+  feature_name="${1}"
+  feature_state="${2}"
+  shift 2
+  feature_modules="${*}"
+
+  if [ -z "${feature_state}" ]; then
+    # shellcheck disable=SC2086
+    if check_for_module ${feature_modules}; then
+      enable_feature "${feature_name}" 1
+    else
+      enable_feature "${feature_name}" 0
+    fi
+  else
+    enable_feature "${feature_name}" "${feature_state}"
+  fi
+}
+
+prepare_cmake_options() {
+  NETDATA_CMAKE_OPTIONS="-S ./ -B ${NETDATA_BUILD_DIR} ${CMAKE_OPTS} ${NETDATA_PREFIX:+-DCMAKE_INSTALL_PREFIX=${NETDATA_PREFIX}} ${NETDATA_USER:+-DNETDATA_USER=${NETDATA_USER}} ${NETDATA_CMAKE_OPTIONS} "
```

Comment:
> ${NETDATA_PREFIX:+-DCMAKE_INSTALL_PREFIX=${NETDATA_PREFIX}}

This change breaks installation because `CMAKE_INSTALL_PREFIX` is no longer set when not using `--install-prefix/--install-no-prefix`. `CMAKE_INSTALL_PREFIX` defaults to `/usr/local` ([docs](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html#cmake-install-prefix)). I am not sure how that was missed.


I guess that is why you added `--install-no-prefix` to Dockerfile 😄

Reply:
The installer script should always have NETDATA_PREFIX set, so the only place this is an issue is in the coverity scan script, but I’m pretty sure that it just doesn’t matter for Coverity.

Reply:
Coverity doesn't matter, what matters is the cmake options - you don't add `CMAKE_INSTALL_PREFIX` when **not** using `--install-prefix/--install-no-prefix`.

Reply:
> ${NETDATA_PREFIX:+-DCMAKE_INSTALL_PREFIX=${NETDATA_PREFIX}}

> If parameter is null or unset, nothing is substituted, otherwise the expansion of word is substituted


nothing is substituted => no `CMAKE_INSTALL_PREFIX` => defaults to `/usr/local`

Reply:
Just realized I was misreading the parameter expansion and that there was a typo in it. It was intended to just use `+`, not `:+`, which would make it equivalent to what was already in the installer script for the installer script, but still ensure that it doesn’t throw an error in the coverity script.